File: ios/ExpoBarcodeScannerHighlightModule.swift
import ExpoModulesCore

public class ExpoBarcodeScannerHighlightModule: Module {
    public func definition() -> ModuleDefinition {
        Name("ExpoBarcodeScannerHighlight")

        View(ExpoBarcodeScannerHighlightView.self) {
            Events("onBarcodesDetected", "onBarcodeTapped")

        }

        Prop("showHighlight") {
            (view: ExpoBarcodeScannerHighlightView, showHighlight: Bool?) in
            view.showHighlight = showHighlight ?? true
        }

        Prop("lerpingSmoothingFactor") {
            (
                view: ExpoBarcodeScannerHighlightView,
                lerpingSmoothingFactor: CGFloat?
            ) in
            view.lerpingSmoothingFactor = lerpingSmoothingFactor ?? 0.3
        }
    }
}

File: android/src/main/java/expo/modules/barcodescannerhighlight/ExpoBarcodeScannerHighlightModule.kt
package expo.modules.barcodescannerhighlight

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ExpoBarcodeScannerHighlightModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ExpoBarcodeScannerHighlight")

    View(ExpoBarcodeScannerHighlightView::class) {
      Events("onBarcodesDetected", "onBarcodeTapped")

      // Prop("showHighlight") { view: ExpoBarcodeScannerHighlightView, showHighlight: Boolean ->
      //   view.showHighlight = showHighlight
      // }

      // Prop("lerpingSmoothingFactor") { view: ExpoBarcodeScannerHighlightView, factor: Double ->
      //   view.lerpingSmoothingFactor = factor.toFloat()
      // }
    }
  }
}